inst/doc/shared_cache.Rmd

```diff
@@ -1,52 +1,191 @@
+
+  <data>temp</data> = readRDS(sc_retrieve(<ind>input_ind_file</ind>))
+  
+  <data>temp_sites</data> <- <data>temp</data> %>%
+    dplyr::filter(
+      dateTime > as.POSIXct('2012-01-01')) %>%
+    dplyr::select(site_no, dateTime) 
+    
+  data_file <- as_data_file(<ind>ind_file</ind>)
+  saveRDS(<data>temp_sites</data>, data_file)
+  gd_put(remote_ind = <ind>ind_file</ind>, local_source = <data>data_file</data>, config_file = <data>gd_config</data>)
+}
+
+</pre> 
 
-Disadvantages of a shared cache (as currently implemented):
-* In a fast-paced collaborative development environment (e.g., a 'sprint'), it is challenging to maintain synchrony between the shared cache (the data) and the git repository (the metadata). Asynchrony is not a deal-breaker but does lead to more rebuilding than would be required for a slower-paced project.
-* Though we've done much to ensure this doesn't happen, it's conceivable that metadata will become corrupt relative to the data. Some monitoring and very occasional full rebuilding is recommended when practical.
-* Old files no longer referenced by the code can accumulate on the shared cache unless manually deleted. Though these will not interfere with ongoing analysis, they can take up storage space unnecessarily.
 
 ### Guidelines
 
-Projects using a shared cache should follow these guidelines:
+####Projects using a shared cache should follow these guidelines:
+
+* Use <ind>**indicator files**</ind> (usually an `.ind` suffix) to represent most or all of the chain of connected <target>**targets**</target> within your remake files. Each `.ind` file should be one of two products of a recipe (e.g. R function call), where the other product is the creation of a <data>**data file**</data>, either locally and/or in the shared cache. The <ind>**indicator files**</ind> are the only products that are declared to `remake`, while the <data>**data files**</data> remain hidden. This allows the <ind>**indicator files**</ind> to indirectly represent the <data>**data files**</data> to `remake` thereby enabling compatibility between `remake` and the shared cache. 
+
+* Always build <target>**targets**</target> using `scipiper::scmake()` rather than `remake::make()`. Though the functions are outwardly very similar, `scmake()` maintains an extra layer of metadata that allows multiple users to share a single project <build>**build status**</build> (e.g., "file x.rds.ind is up to date; file y.rds.ind is out of date"). In a shared-cache project, you should not even need to load the `remake` package directly.
+
+* Generally avoid using R objects as `remake/scmake` <target>**targets**</target>...but if you must, usually for convenience or conciseness of the workflow plan, recognize that R objects must be built by every analyst. So if a target takes a non-trivial length of time to build, or if it depends on large volumes of <data>**data**</data> as input, that target should usually be a file rather than an R object.
```

Comment:
Why should you generally avoid this? I have a lot of code that does this - particularly when reading from config files.

Reply:
Yeah, is this the consensus? And if so, I'd also vote to have some explanation why to avoid objects. That was one of my main reasons for preferring `remake` to `make`.

Reply:
But surely you tend to prefer objects over files specifically when they take a trivial length of time to build or depend on less-than-large volumes of data as input (config files being a good example)? Or according to some other considerations?

@wdwatkins is scheduled to do some serious thinking and writing about this for our march 26 pipelines meeting:` David - March 26 - Doc: When to use R objects vs data files vs indicator+data files` - let's make sure we get any related thoughts over to him for that.

Reply:
@aappling-usgs ...I'm not sure if your question is directed at me or Jake...so if me: My preference is to default to objects unless there is a reason to have to deal with files, and I guess if you've committed some target to following the shared cache approach, that would be a reason for a file (?). The sentence made it seem like all scipiper targets should avoid objects.  Maybe all we need to do is "Generally avoid using R objects as shared-cache targets"? 

I'm hoping David's vignette will tease out some benchmarks for a file (file size, sharing with non-R users, shared cache that needs to be pushed up to AWS/GoogleDrive/etc...and *that* probably depends on how long you are willing to wait for computations).

Reply:
(Yes, my comment was directed at Laura and Lindsay's comments.)

Good point that the guidelines for object vs file should be quite different for shared-cache vs non-shared-cache projects (and could be more like non-shared-cache guidelines for specific task tables even within generally-shared-cache projects). I'm also intrigued by the challenge to provide specific benchmarks.

Note that this particular vignette and PR are for a vignette specifically about shared caches.

Reply:
I'll specify in the paragraph that it's focusing on shared cache and also that there may be thresholds / guidelines to file size, etc... for when to do R object vs. file (this can get updated after David's pipelines meeting)